### PR TITLE
filesystem & binaryplatforms: remove undefined vars from exports

### PR DESCRIPTION
**Pull request by timholy**
These are never defined. They show up as noise during the ambiguity test.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Could we add a CI job that fails if there are undefined exports?

I know that Aqua.jl has a similar functionality for package tests. It would be nice to have something like that for Base and stdlibs.

**Comment by timholy** [comment]
We could add that to `detect_ambiguity` (although that doesn't specifically test exports, these simply get flagged as undefined names in `Base`). The one hitch is the handling of `active_repl` and `active_repl_backend`, which are defined as `global`s in `base/client.jl: run_main_repl`.

**Comment by DilumAluthge** [comment]
> We could add that to `detect_ambiguity` (although that doesn't specifically test exports, these simply get flagged as undefined names in `Base`).

That seems like a good idea.

> The one hitch is the handling of `active_repl` and `active_repl_backend`, which are defined as `global`s in `base/client.jl: run_main_repl`.

Ideally, we would fix our "detect undefined names" tooling to correctly detect globals that are defined inside functions.

**Comment by KristofferC** [comment]
Ref https://github.com/JuliaLang/julia/pull/28693. Maybe the test stopped working.

**Comment by timholy** [comment]
Should still work. None of these names were exported by Base, they were exported by internal modules.

**Comment by KristofferC** [comment]
Ah, I see.

File: base/filesystem.jl [hunk 1]
```diff
@@ -45,10 +45,6 @@ export File,
        JL_O_CREAT,
        JL_O_EXCL,
        JL_O_TRUNC,
-       JL_O_TEMPORARY,
-       JL_O_SHORT_LIVED,
-       JL_O_SEQUENTIAL,
-       JL_O_RANDOM,
```
Comment (vtjnash) [inline base/filesystem.jl:51]: These are defined, just perhaps not on your system: https://github.com/JuliaLang/julia/blob/master/src/file_constants.h

**Review by vtjnash — COMMENTED** [review]
This was a breaking change. We can define this to have a value (0o000?), but cannot delete them now.

File: base/filesystem.jl (hunk 1 above)
Comment (vtjnash) [reply base/filesystem.jl:51]: On second thought, we should perhaps switch to the correct names for them (e.g. http://docs.libuv.org/en/v1.x/fs.html#c.UV_FS_O_SEQUENTIAL)

Comment (timholy) [reply base/filesystem.jl:51]: I'll leave the choice to you, and just revert this change (#43817)

**Pull request merged**